### ISSUE TITLE
Icons cut off on the right side

### ISSUE THREAD

**Issue by ziombo** [nvim-neo-tree/neo-tree.nvim#371]
Some (or maybe all) of the icons I'm using are not displayed properly in neo-tree, as they are cut on the right side:
![image](https://user-images.githubusercontent.com/15613606/172069959-7b46f97f-a6ed-49cc-b6cc-94082c4f831f.png)

The same isn't happening for other plugins like bufferline:
![image](https://user-images.githubusercontent.com/15613606/172069984-560d7b65-73e8-4ee6-b0e0-26835b69d52c.png)

I'm using Windows Terminal with WSL2 Ubuntu20.4 and I know that there are some problems with Windows Terminal displaying glyphs that are wider than usual, but since it's working with bufferline I'm wondering if maybe it has something to do with neo-tree?

I tried adding a trailing space to the icon, but then neo-tree doesn't display any icon (bufferline respects the additional space):
![image](https://user-images.githubusercontent.com/15613606/172070118-e8b44d40-acfc-4669-9fbf-70ef2af67915.png)

I did some investing trying to see why the icons are sometimes displayed and sometimes not and I believe that if an icon is more than 3 chars, e.g. icon + 2 trailing spaces, neo-tree wont display it (unlike bufferline). Also, neo-tree doesn't seem to be respecting the trailing space at all, since an icon with trailing space is displayed in a same way that the one without.

**Comment by cseickel** [comment]
I also use Windows Terminal sometimes and I have seen this inn certain situations but not all the time. It has to do with the font, font size, and specific characters you are using. It might also have to do with whether you are using Windows Terminal Preview or the normal release.

It's possible that it's being causing by the string concat and slicing that is being don't to right align, because lua does not handle wide characters natively. I'll take a look this week.

**Comment by gsalvatella** [comment]
I can confirm this as well. Some fonts work particularly bad while others display the signs correctly.

**Comment by ur4ltz** [comment]
Same problem, only I noticed that the icon just goes outside the window and appears temporarily when the window width is changed.

**Comment by cseickel** [comment]
> Slackware Linux v14.2 x86, terminal xterm.

This was helpful because I do all of my personal dev work on linux and I can actually recreate the issue now. I will try to fix this.

In the meantime, you should know that it works fine in Kitty.

**Comment by cseickel** [comment]
I think this is actually fixed now, at least for xterm. Please let me know if it is foxed for everyone. **These changes are in the `main` branch right now.**

For anyone else that comes across a similar problem, here is what I _think_ is going on. You need to have a space after a double width character to give it room to display, but how you add that space matters. It needs to be right after the characters and within the same highlight group. If the highlight group ends, and a new one starts, it will be cut off, even if that new highlight group has the same background and contains only a space.

**Comment by pysan3** [comment]
Yes! Switching to the `main` branch fixed the problem in my setup.

Thank you very much!!

- WSL, Windows Terminal, font: `PlemolJP` (extended font of `IBM Plex Mono`)

**Comment by ur4ltz** [comment]
Yes! Commit 365df55 fixed the problem for me.

**Comment by ziombo** [comment]
Fixed it for me as well
